rust/operator-binary/src/config/fault_tolerant_execution.rs

```diff
@@ -0,0 +1,505 @@
+use std::collections::BTreeMap;
+
+use snafu::{ResultExt, Snafu};
+use stackable_operator::{
+    builder::pod::volume::{VolumeBuilder, VolumeMountBuilder},
+    client::Client,
+    crd::s3,
+    k8s_openapi::{
+        api::core::v1::{Volume, VolumeMount},
+        apimachinery::pkg::api::resource::Quantity,
+    },
+};
+
+use crate::{
+    config,
+    crd::{
+        CONFIG_DIR_NAME,
+        fault_tolerant_execution::{
+            ExchangeManagerBackend, FaultTolerantExecutionConfig, HdfsExchangeConfig,
+        },
+    },
+};
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("Failed to resolve S3 connection"))]
+    S3Connection {
+        source: s3::v1alpha1::ConnectionError,
+    },
+
+    #[snafu(display("Failed to resolve S3 connection"))]
+    ResolveS3Connection { source: config::s3::Error },
+
+    #[snafu(display("trino does not support disabling the TLS verification of S3 servers"))]
+    S3TlsNoVerificationNotSupported,
+
+    #[snafu(display("failed to convert data size for [{field}] to bytes"))]
```

Comment:
just FYI that this PR has multiple inconsistent starting casing. "The industry" tends toward lowercase AFAIK.
No need to change anything, just for the future